packages/realtime_client/lib/src/realtime_client.dart

```diff
@@ -387,13 +407,17 @@ class RealtimeClient {
 
   /// communication has been closed
   void _onConnClose() {
-    final event = conn?.closeReason ?? '';
+    final statusCode = conn?.closeCode;
+    RealtimeCloseEvent? event;
+    if (statusCode != null) {
+      event = RealtimeCloseEvent(code: statusCode, reason: conn?.closeReason);
+    }
```

Comment:
JS client passes [ClosedEvent](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent) when the connection is closed, so I added something similar here.

Reply:
Also wanted to share for reference that `conn.closeReason` was often an empty string where as `closeCode` had values whenever the connection is closed, so it should be more helpful.